packages/ckeditor5-engine/src/conversion/upcastdispatcher.js

```diff
@@ -494,10 +495,13 @@ function createContextTree( contextDefinition, writer ) {
 			attributes[ key ] = item.getAttribute( key );
 		}
 
-		const current = writer.createElement( item.name, attributes );
+		let current;
 
 		if ( position ) {
+			current = writer.createElement( item.name, attributes );
 			writer.append( current, position );
+		} else {
+			current = new RootElement( null, item.name );
```

Comment:
I don't like that we pass `null` here.

Besides that, we need more information about this change. Why is it needed. Why are `Title`'s tests failing without it. What's the impact. Etc.

Reply:
It looks like reverting this change will work with `element.is( 'element', '$root' )` - this, IMO, showcase another problem with the check for is root. I've asked @niegowski to describe it in a new issue. 

There are some shortcuts in current code which checked for root element like `element.is( '$root' )` and schema context creates default elements using `createElement( name )` which previously worked and with `element.is( 'rootElement', '$root' )` would fail.

Reply:
issue created: #7663